src/scripts.js

```diff
@@ -70,7 +70,13 @@ function isScriptHashInput(script, allowIncomplete) {
   if (!Buffer.isBuffer(lastChunk)) return false
 
   var scriptSig = Script.fromChunks(script.chunks.slice(0, -1))
-  var scriptPubKey = Script.fromBuffer(lastChunk)
+  var scriptPubKey
+
+  try {
+    scriptPubKey = Script.fromBuffer(lastChunk)
+  } catch (e) {
```

Comment:
I was wondering the same. How about catching all for now and `console.debug` the exception just in case someone may wonder at which stage the classification returned. Hopefully they report back if it's an unexpected case.